Timeline of loadgen stats

mhofman opened the pull request:
This PR provides the plumbing to generate a CSV file containing a small subset of the loadgen stats across multiple revisions.

The file is currently generated locally and manually imported into a google spreadsheet. I still want to figure out a better way, by uploading the results automatically somewhere, and make them available to a hosted dashboard, possibly through a "revision completion" hook for the background script.

I generated a preliminary corpus of data on my own machine, with the results available [here](https://docs.google.com/spreadsheets/d/174eYWAlunqVm0ENXfJWuFBNvIsi6UPX4KqPp4SFFykI/edit#gid=1343630455), using the following config: `--no-stage.save-storage --monitor-interval=1 --stage.duration=10 --stage.loadgen.vault.interval=30 --stage.loadgen.amm.interval=30 --stage.loadgen.amm.wait=15 --stage.loadgen.vault.limit=2 --stage.loadgen.amm.limit=2`

A longer version is currently being generated on the benchmark machine for the same revisions with the following config: `--no-stage.save-storage --monitor-interval=1 --stage.duration=30 --stage.loadgen.vault.interval=60 --stage.loadgen.amm.interval=60 --stage.loadgen.amm.wait=30`

I'd ultimately like to run the default 24h test in the background on a subset of past revisions to backfill the data, but I'm not sure which revisions would be the best candidates yet.

Stack on top of compatibility fixes (#37 / #44) and stats work (#36).

As always, best reviewed commit-by-commit.

michaelfig commented on Dockerfile:
```diff
@@ -150,12 +150,13 @@ ENV IS_DOCKER=true
 ENV SDK_SRC=/src
 ENV OUTPUT_DIR=/out
 ENV SDK_REVISION=
+Env SDK_REPO=
```
This is the first time I've seen a non-capitalised Docker command.  Interesting.

mhofman replied:
Resolved in 41242a65e6bbd5660c7df1262e05ec069b09e022

michaelfig commented on Dockerfile:
```diff
@@ -150,12 +150,13 @@ ENV IS_DOCKER=true
 COPY --chown=$USER_UID:$USER_GID . .
 
-RUN mkdir -p $SDK_SRC $OUTPUT_DIR && chown $USER_UID:$USER_GID $SDK_SRC $OUTPUT_DIR /app
+RUN mkdir -p $SDK_SRC $OUTPUT_DIR /home/node/.cache/yarn /go/pkg/mod && chown -R $USER_UID:$USER_GID $SDK_SRC $OUTPUT_DIR /home/node/.cache /go && chown $USER_UID:$USER_GID /app
```
Please split long lines.

```suggestion
RUN mkdir -p $SDK_SRC $OUTPUT_DIR /home/node/.cache/yarn /go/pkg/mod && \
  chown -R $USER_UID:$USER_GID $SDK_SRC $OUTPUT_DIR /home/node/.cache /go && \
  chown $USER_UID:$USER_GID /app
```

mhofman replied:
Resolved in 6287b8096ebd48bb08671feb6c24c422e917dd3f

michaelfig commented on scripts/common-queue.sh:
```diff
@@ -0,0 +1,47 @@
+    done
+    echo "processing ${SDK_REVISION}"
+    mkdir "${OUTPUT_DIR}"
+    DOCKER_ID=$(docker create -v loadgen-go-pkg-mod:/go/pkg/mod -v loadgen-yarn-cache:/home/node/.cache/yarn -v "$(pwd)/${OUTPUT_DIR}:/out" -e SDK_REVISION=${SDK_REVISION} -e SDK_REPO=${SDK_REPO} --name "${OUTPUT_DIR}" loadgen-runner --test-data.test-type=${TEST_TYPE} "$@") || exit $?
```
```suggestion
    DOCKER_ID=$(docker create -v loadgen-go-pkg-mod:/go/pkg/mod \
      -v loadgen-yarn-cache:/home/node/.cache/yarn \
      -v "$(pwd)/${OUTPUT_DIR}:/out" \
      -e SDK_REVISION=${SDK_REVISION} \
      -e SDK_REPO=${SDK_REPO} \
      --name "${OUTPUT_DIR}" \
      loadgen-runner \
      --test-data.test-type=${TEST_TYPE} "$@") || exit $?
```

mhofman replied:
Resolved in 6287b8096ebd48bb08671feb6c24c422e917dd3f

michaelfig commented on scripts/README.md:
````diff
@@ -0,0 +1,29 @@
+### Manual perf example
+
+```sh
+cat << EOF | /path/to/testnet-load-generator/scripts/run-manual.sh - --no-stage.save-storage --stage.duration=30 --monitor-interval=1 --stage.loadgen.vault.interval=60 --stage.loadgen.amm.interval=60 --stage.loadgen.amm.wait=30
````
```suggestion
cat << EOF | /path/to/testnet-load-generator/scripts/run-manual.sh - \
  --no-stage.save-storage \
  --stage.duration=30 \
  --monitor-interval=1 \
  --stage.loadgen.vault.interval=60 \
  --stage.loadgen.amm.interval=60 \
  --stage.loadgen.amm.wait=30
```

mhofman replied:
Resolved in f0b5f1a8953212b6427cbb93c9367a7f83be9d33

michaelfig commented on scripts/perf_to_stats_csv.jq:
```diff
@@ -0,0 +1,32 @@
+  .stats |
+  {
+    sdkRevision: (.metadata.testData.sdkRevision | tostring), 
+    sdkCommitTime: (.metadata.testData.sdkCommitTime | strftime("%Y-%m-%d %H:%M:%S")),
```
What timezone is the sdkCommitTime measured in?  I'd prefer UTC, if possible.

mhofman replied:
This is plumbed from the `git show` command producing a Unix timestamp:
https://github.com/Agoric/testnet-load-generator/blob/d8c3e286349ca43d7ac64aac1a3bfb1bf7109295/start.sh#L37

And according to [`jq`'s manual](https://stedolan.github.io/jq/manual/#Dates), `strftime` formats using UTC/GMT: 
> The `strftime(fmt)` builtin formats a time (GMT) with the given format. The `strflocaltime` does the same, but using the local timezone setting.

michaelfig reviewed (approved):
LGTM!  I have just a few suggestions, but you're welcome to rebut them if you don't find them useful.

mhofman reviewed (commented):
Thanks for catching those. I agree, splitting long lines is more readable. I need a prettier for bash, I've become to lazy to format my lines manually.

warner reviewed (approved):
nice

mhofman commented:
@michaelfig since your initial review I addressed your feedback and added d30dea34b014f0fbd817bf6fcd5acdcede5652e1 and 8717013aaa8b639cfc512fe1a9fbfd2a89328efd.

Since those are minor changes, and @warner approved the PR following them, I will squash the feedback commits, and merge.

mhofman commented:
Scripts to generate a CSV file containing a small subset of the loadgen stats across multiple revisions.
Increase default loadgen load
